robot/impl/local_robot_test.go

```diff
@@ -3482,3 +3482,21 @@ func TestCustomResourceBuildsOnModuleAddition(t *testing.T) {
 	)
 	rtestutils.VerifySameResourceNames(t, r.ResourceNames(), expectedResources)
 }
+
+func TestSendTriggerConfig(t *testing.T) {
+	// Tests that the triggerConfig channel buffers exactly 1 message and that
+	// sendTriggerConfig does not block whether or not the channel has
+	// capacity.
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	r := setupLocalRobot(t, ctx, &config.Config{}, logger)
+	r.Close(ctx)
```

Comment:
why do we need to close here?

Reply:
the close stops the reconfigureWorker, so closing it here means that no messages will be processed. the test is checking that sendTriggerConfig is attempting to send messages to the channel but not blocking if it fails. That's why there's a check that there is exactly one message in the `triggerConfig` channel

Reply:
Won't this piece of `sendTriggerConfig` stop _any_ of the 5 calls from sending anything to the channel?
```
if r.closeContext.Err() != nil {
	return
}
```

Reply:
that's why I re-initialize the context [here](https://github.com/viamrobotics/rdk/pull/3968/files#diff-d44b43f9f1e7b665af6465a60419c6684582f59225bcde5e60d7931932895c60R3497).

Reply:
Oops totally missed that; thanks.

Reply:
also added more comments in the test